rust/crd/src/authentication.rs

```diff
@@ -0,0 +1,290 @@
+    pub hostname: String,
+    /// Port of the LDAP server
+    #[serde(default = "default_port")]
+    pub port: u16,
+    /// LDAP search base
+    #[serde(default)]
+    pub search_base: String,
+    /// LDAP query to filter users
+    #[serde(default)]
+    pub search_filter: String,
+    /// The name of the username field
+    #[serde(default = "default_uid_field")]
+    pub uid_field: String,
+    /// The name of the group field
+    #[serde(default = "default_group_field")]
+    pub group_field: String,
+    /// The name of the firstname field
+    #[serde(default = "default_firstname_field")]
+    pub firstname_field: String,
+    /// The name of the lastname field
+    #[serde(default = "default_lastname_field")]
+    pub lastname_field: String,
+    /// The name of the email field
+    #[serde(default = "default_email_field")]
+    pub email_field: String,
+    /// In case you need a special account for searching the LDAP server you can specify it here
+    pub bind_credentials: Option<AuthenticationClassLdapBindCredentials>,
+    /// Use a TLS connection. If not specified no TLS will be used
+    pub tls: Option<AuthenticationClassTls>,
+}
+
+fn default_port() -> u16 {
+    389
+}
+
+fn default_uid_field() -> String {
+    "uid".to_string()
+}
+
+fn default_group_field() -> String {
+    "memberof".to_string()
+}
+
+fn default_firstname_field() -> String {
+    "givenName".to_string()
+}
+
+fn default_lastname_field() -> String {
+    "sn".to_string()
+}
+
+fn default_email_field() -> String {
+    "mail".to_string()
+}
+
+#[derive(Clone, Debug, Deserialize, Eq, JsonSchema, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct AuthenticationClassLdapBindCredentials {
+    /// [SecretClass](https://docs.stackable.tech/secret-operator/secretclass.html) containing the LDAP bind credentials
+    pub secret_class: String,
+    /// [Scope](https://docs.stackable.tech/secret-operator/scope.html) of the [SecretClass](https://docs.stackable.tech/secret-operator/secretclass.html)
+    pub scope: Option<AuthenticationClassSecretClassScope>,
+}
+
+#[derive(Clone, Debug, Deserialize, Eq, JsonSchema, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct AuthenticationClassSecretClassScope {
+    #[serde(default)]
+    pub pod: bool,
+    #[serde(default)]
+    pub node: bool,
+    #[serde(default)]
+    pub services: Vec<String>,
+}
+
+#[derive(Clone, Debug, Deserialize, Eq, JsonSchema, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub enum AuthenticationClassTls {
+    /// Use TLS but don't verify certificates.
+    /// We have to use an empty struct instead of an empty Enum because of limitations of [kube-rs](https://github.com/kube-rs/kube-rs/)
+    Insecure {},
+    /// Use TLS and the ca certificates provided by the system - in this case the Docker image - to verify the server.
+    /// This can be useful when you e.g. use public AWS S3 or other public available services.
+    SystemProvided {},
```

Comment:
This doesn't seem very relevant to LDAP. Maybe we can drop this for now and come back to it later if a legitimate use case arises?